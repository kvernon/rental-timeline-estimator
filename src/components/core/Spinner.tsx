import styled from '@emotion/styled';
import { IShape } from './IShape';

const shapeSize = 950;

export const Spinner = styled.div<{ shape: IShape }>`
  @keyframes spin {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }

  width: ${shapeSize}px;
  height: ${shapeSize}px;
  left: ${(props) => {
    if (props.shape.x && props.shape.width) {
      return props.shape.x + props.shape?.width / 2 - shapeSize / 2;
    }
    return 240;
  }}px;
  top: -332px;
  float: left;
  position: absolute;
  background-image: radial-gradient(rgba(199, 159, 0, 0.3) 0%, rgba(61, 61, 61, 0.01) 62%, transparent 67%);
  border-radius: 50%;
  flex-grow: 1;
  z-index: -1;
  animation-name: spin;
  animation-duration: 300s;
  animation-iteration-count: infinite;
  animation-timing-function: linear;

  &::after,
  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    transform-origin: center;
    border-radius: 50%;
    mask-image: radial-gradient(rgba(0, 0, 0, 1) 65%, transparent 80%);
  }

  &::before {
    background: repeating-conic-gradient(from 0deg, rgb(199, 159, 0, 0.3) 0deg 10deg, transparent 10deg 15deg);
    mask-image: radial-gradient(rgba(0, 0, 0, 0.7) 0%, rgba(0, 0, 0, 0.3) 36%, transparent 65%);
    animation:
      rotate 720s linear,
      scale 3s linear infinite;
  }
`;
