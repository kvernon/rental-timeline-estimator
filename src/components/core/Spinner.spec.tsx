import React from 'react';
import { configure, render, screen } from '@testing-library/react';
import { Spinner } from './Spinner';

describe('Spinner unit tests', () => {
  beforeEach(() => {
    configure({ testIdAttribute: 'id' });
  });
  test('should render with shape', () => {
    render(<Spinner role="spinner" shape={{ x: 1, width: 30, y: 2 }} />);

    const entity = screen.getByRole<HTMLDivElement>('spinner');

    expect(entity).toMatchSnapshot();
  });
  test('should render without shape', () => {
    render(<Spinner role="spinner" shape={{ x: undefined, width: undefined, y: undefined }} />);

    const entity = screen.getByRole<HTMLDivElement>('spinner');

    expect(entity).toMatchSnapshot();
  });
});
