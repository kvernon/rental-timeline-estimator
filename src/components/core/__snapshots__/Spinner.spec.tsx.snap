// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner unit tests should render with shape 1`] = `
.emotion-0 {
  width: 950px;
  height: 950px;
  left: -459px;
  top: -332px;
  float: left;
  position: absolute;
  background-image: radial-gradient(rgba(199, 159, 0, 0.3) 0%, rgba(61, 61, 61, 0.01) 62%, transparent 67%);
  border-radius: 50%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  z-index: -1;
  -webkit-animation-name: spin;
  animation-name: spin;
  -webkit-animation-duration: 300s;
  animation-duration: 300s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.emotion-0::after,
.emotion-0::before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  transform-origin: center;
  border-radius: 50%;
  -webkit-mask-image: radial-gradient(rgba(0, 0, 0, 1) 65%, transparent 80%);
  mask-image: radial-gradient(rgba(0, 0, 0, 1) 65%, transparent 80%);
}

.emotion-0::before {
  background: repeating-conic-gradient(from 0deg, rgb(199, 159, 0, 0.3) 0deg 10deg, transparent 10deg 15deg);
  -webkit-mask-image: radial-gradient(rgba(0, 0, 0, 0.7) 0%, rgba(0, 0, 0, 0.3) 36%, transparent 65%);
  mask-image: radial-gradient(rgba(0, 0, 0, 0.7) 0%, rgba(0, 0, 0, 0.3) 36%, transparent 65%);
  -webkit-animation: rotate 720s linear,scale 3s linear infinite;
  animation: rotate 720s linear,scale 3s linear infinite;
}

<div
  class="emotion-0"
  role="spinner"
  shape="[object Object]"
/>
`;

exports[`Spinner unit tests should render without shape 1`] = `
.emotion-0 {
  width: 950px;
  height: 950px;
  left: 240px;
  top: -332px;
  float: left;
  position: absolute;
  background-image: radial-gradient(rgba(199, 159, 0, 0.3) 0%, rgba(61, 61, 61, 0.01) 62%, transparent 67%);
  border-radius: 50%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  z-index: -1;
  -webkit-animation-name: spin;
  animation-name: spin;
  -webkit-animation-duration: 300s;
  animation-duration: 300s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.emotion-0::after,
.emotion-0::before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  transform-origin: center;
  border-radius: 50%;
  -webkit-mask-image: radial-gradient(rgba(0, 0, 0, 1) 65%, transparent 80%);
  mask-image: radial-gradient(rgba(0, 0, 0, 1) 65%, transparent 80%);
}

.emotion-0::before {
  background: repeating-conic-gradient(from 0deg, rgb(199, 159, 0, 0.3) 0deg 10deg, transparent 10deg 15deg);
  -webkit-mask-image: radial-gradient(rgba(0, 0, 0, 0.7) 0%, rgba(0, 0, 0, 0.3) 36%, transparent 65%);
  mask-image: radial-gradient(rgba(0, 0, 0, 0.7) 0%, rgba(0, 0, 0, 0.3) 36%, transparent 65%);
  -webkit-animation: rotate 720s linear,scale 3s linear infinite;
  animation: rotate 720s linear,scale 3s linear infinite;
}

<div
  class="emotion-0"
  role="spinner"
  shape="[object Object]"
/>
`;
